content/blog/apl/2020-07-11_repository_identification.md

```diff
@@ -0,0 +1,99 @@
++++
+type = "blog"
+title = "Haiku Repository Files and Identifiers"
+author = "apl"
+date = "2020-07-11 11:42:49+12:00"
+tags = ["haiku", "software", "repository"]
++++
+
+Software on a computing platform such as Haiku is typically distributed as a package.  Without a packaging system it would be hard for users to install software and because software often depends on other software, the chain of dependencies would be  difficult for a user to resolve themselves.  To orchestrate the distribution and management of the packages, Haiku has a packaging system which consists of applications, online tools, on-host tools and software libraries.  One aspect of the packaging system is the coordination and identification of repositories.
+
+## What is Stored in a Repository?
+
+A repository is a collection of packages coupled with an index of those packages.
+
+A repository could reside on any storage media.  This is most conveniently a web-server, but could also be a DVD for example.  It is simply a collection of files.  The `baseUrl` is used to identify the *transport* mechanism and address used to access the repository data.  The `baseUrl` is what users require to configure a repository on their Haiku system.  By way of example, the `baseUrl` of HaikuPorts repository for the x86_64 computing architecture is;
+
+```
+https://eu.hpkg.haiku-os.org/haikuports/master/x86_64/current
+```
+
+The repository also contains a special file `repo.info` that describes the repository found at a given `baseUrl`.  Here is an example of what this might look like;
+
+```
+name HaikuPorts
+vendor "Haiku Project"
+summary "The HaikuPorts repository"
+priority 1
```

Comment:
Can you add a very quick explanation of the "priority" ?

Reply:
Sorry I'm not too sure what the intent of this is and I can't see how it currently has any bearing on the behaviour in the code.

Reply:
I suggest not to advertise our ignorance then. :)